crates/findex/src/pyo3/py_structs.rs

```diff
@@ -1,106 +1,68 @@
-use std::{
-    collections::hash_map::DefaultHasher,
-    hash::{Hash, Hasher},
-};
+use std::hash::Hash;
 
 use cosmian_crypto_core::{reexport::rand_core::SeedableRng, CsRng};
 use cosmian_findex::{
     parameters::MASTER_KEY_LENGTH, IndexedValue as IndexedValueRust,
-    KeyingMaterial as KeyingMaterialRust, Keyword, Label as LabelRust, Location,
+    KeyingMaterial as KeyingMaterialRust, Keyword as KeywordRust, Label as LabelRust,
+    Location as LocationRust,
 };
 use pyo3::{prelude::*, pyclass::CompareOp, types::PyBytes};
 
-/// The value indexed by a `Keyword`. It can be either a `Location` or another
-/// `Keyword` in case the searched `Keyword` was a tree node.
-#[pyclass]
-#[derive(Hash, PartialEq, Eq, Clone)]
-pub struct IndexedValue(pub(super) IndexedValueRust);
-
-#[pymethods]
-impl IndexedValue {
-    /// Create `IndexedValue` from a location in bytes.
-    ///
-    /// Args:
-    ///     location_bytes (bytes)
-    ///
-    /// Returns:
-    ///     IndexedValue
-    #[staticmethod]
-    pub fn from_location(location_bytes: &[u8]) -> Self {
-        Self(IndexedValueRust::Location(Location::from(location_bytes)))
-    }
-
-    /// Create `IndexedValue` from a keyword in bytes.
-    ///
-    /// Args:
-    /// keyword_bytes (bytes)
-    ///
-    /// Returns:
-    ///     IndexedValue
-    #[staticmethod]
-    pub fn from_keyword(keyword_bytes: &[u8]) -> Self {
-        Self(IndexedValueRust::NextKeyword(Keyword::from(keyword_bytes)))
+fn truncate(s: String, max_chars: usize) -> String {
+    match s.char_indices().nth(max_chars) {
+        None => s,
+        Some((idx, _)) => format!("{}...", &s[..idx]),
     }
+}
```

Comment:
A function already exists in the standard implementation of `String` that does what you want but without copy (it drops terminating chars without reallocating):
```suggestion
fn truncate(s: String, max_chars: usize) -> String {
    s.truncate(max_chars);
    s
}
```

Reply:
You could have truncated to `max_char + 3` then rewrite the last three chars by `.`